### ISSUE TITLE
errors do not terminate source unless an observer is provided

### ISSUE THREAD

**Issue by colinfang** [ReactiveX/rxjs#1921]
```
Observable.interval(1000)
  .flatMap(() => this.http.get('abc'))
  .subscribe(
      // x => console.log('Next: ' + x) // Don't terminite without this.
  );
```

This http request gives an error due to 404. But the stream keeps polling.

However, as soon as I uncomment the commented line in subscribe, it terminates on the first error.

See example here

http://plnkr.co/edit/bQyBvboFJCPyZUEFjraX?p=preview

Why is that?

**Comment by jayphelps** [comment]
Appears this isn't related to `flatMap`, it's more general than that. If you don't provide an observer to `subscribe(observer)` then the source does not terminate correctly.

**Comment by jayphelps** [comment]
Still digging, but here's my minimal example:

http://jsbin.com/yayave/edit?js,output

So far my best guess is it's related to the internal `syncErrorThrowable`, whether it's true or false. Not providing an observer it gets the default `empty` observer which is `syncErrorThrowable: false` but when you provide your own its `syncErrorThrowable: true`. This logic is seemingly for optimizing try/catch.

@blesh this prolly needs your attention since I believe you're the one who knows this part best.

**Comment by trxcllnt** [comment]
@jayphelps I'm responsible for `syncErrorThrowable

**Comment by jayphelps** [comment]
@trxcllnt this is all YOUR FAULT!

**Comment by trxcllnt** [comment]
@jayphelps I may have authored the code, but `syncErrorThrowable` is the product of a debate on error handling semantics that [I lost](https://github.com/tc39/proposal-observable/issues/47), so I'm going to blame @blesh and @zenparsing :-)

**Comment by trxcllnt** [comment]
@jayphelps @colinfang fixed in #1935
